### PR TITLE
[Maintenance]: will now obtain the gemloader scripts per CTP7 version

### PR DESCRIPTION
**Pull request by bdorney**
## Description

Removing hard coded `wget` to `3.5.0` gemloader scripts and now using the per tag scripts in the `GEM_AMC` repo.

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Motivation and Context

Previously we were using hard coded version to the 3.5.0 release.  Now we will always get the scripts from the tag associated with the release (unless @evka85 can ensure each release would include these files; which has not been the case thus far...relying on the tag seems to be more reliable).

## How Has This Been Tested?

Tested here: http://cmsonline.cern.ch/cms-elog/1086736

This PR reflects state after solving all issues.  Note that LMDB needs to still be updated by hand since the version of the libxhal_ctp7.so library in xhal tag 3.2.2 is horrifically out of date.

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: setup_ctp7.sh [hunk 1]
```diff
@@ -161,15 +161,18 @@ then
...
+        mkdir gemloader
+    fi
+    echo "wget https://raw.githubusercontent.com/evka85/GEM_AMC/v${ctp7fw//./_}/scripts/gemloader/gemloader_clear_header.sh -P gemloader/"
+    wget https://raw.githubusercontent.com/evka85/GEM_AMC/v${ctp7fw//./_}/scripts/gemloader/gemloader_clear_header.sh -P gemloader/
+    echo "wget https://raw.githubusercontent.com/evka85/GEM_AMC/v${ctp7fw//./_}/scripts/gemloader/gemloader_configure.sh -P gemloader/"
+    wget https://raw.githubusercontent.com/evka85/GEM_AMC/v${ctp7fw//./_}/scripts/gemloader/gemloader_configure.sh -P gemloader/
+    echo "wget https://raw.githubusercontent.com/evka85/GEM_AMC/v${ctp7fw//./_}/scripts/gemloader/gemloader_load_test_data.sh -P gemloader/"
+    wget https://raw.githubusercontent.com/evka85/GEM_AMC/v${ctp7fw//./_}/scripts/gemloader/gemloader_load_test_data.sh -P gemloader/
+    echo "wget https://raw.githubusercontent.com/evka85/GEM_AMC/v${ctp7fw//./_}/scripts/gemloader/gemloader_read.sh -P gemloader/"
+    wget https://raw.githubusercontent.com/evka85/GEM_AMC/v${ctp7fw//./_}/scripts/gemloader/gemloader_read.sh -P gemloader/
```
Comment (jsturdy) [inline setup_ctp7.sh]: All of these are coming from `v${ctp7fw//./_}`, no, just put the file names into an array and do a loop? (probably also for subsequent lines)

**Review by jsturdy — CHANGES_REQUESTED** [review]

File: setup_ctp7.sh (hunk 1 above)
Comment (bdorney) [reply setup_ctp7.sh]: yeah that can be done.

Comment (bdorney) [reply setup_ctp7.sh]: Resolved; an array is now used.

**Comment by bdorney** [comment]
@mexanick and @evka85 review re-requested.

File: scripts/gth_config_opto.sh [hunk 2]
```diff
@@ -31,7 +31,7 @@ do
 	then
 
 		printf "Applying special inverted RX channel config for GTH CH#$i...\n"
-		mpoke $((gth_ctrl_ch0_base_addr+256*11)) $data_ch_inverted
+		mpoke $gth_ctrl_addr $data_ch_inverted
```
Comment (jsturdy) [inline scripts/gth_config_opto.sh:34]: This was a change you made, or that came as a result of copying newer scripts into the repo?
(Is this one of the eponymous "scripts"?)

File: scripts/gth_config_opto_tx_inverted.sh [hunk 3]
```diff
@@ -31,7 +31,7 @@ do
 	then
 
 		printf "Applying special inverted RX channel config for GTH CH#$i...\n"
-		mpoke $((gth_ctrl_ch0_base_addr+256*11)) $data_ch_inverted
+		mpoke $gth_ctrl_addr $data_ch_inverted
```
Comment (jsturdy) [inline scripts/gth_config_opto_tx_inverted.sh:34]: Similarly

File: scripts/gth_config_opto.sh (hunk 2 above)
Comment (bdorney) [reply scripts/gth_config_opto.sh:34]: from copying newer scripts.

File: scripts/gth_config_opto_tx_inverted.sh (hunk 3 above)
Comment (bdorney) [reply scripts/gth_config_opto_tx_inverted.sh:34]: from copying newer scripts.

**Review by jsturdy — APPROVED** [review]
The scripts are copied into this repository because they aren't (currently) attached to the FW release, correct?

**Comment by bdorney** [comment]
> The scripts are copied into this repository because they aren't (currently) attached to the FW release, correct?

that's correct.

**Pull request merged**